### DISCUSSION TITLE
Dirsearch as a completely Client-Server App?

### DISCUSSION THREAD

**Discussion by maurosoria** [maurosoria/dirsearch#867] [Ideas]
Hello folks! How you doing? Hope you are all safe in these crazy days.

Dirsearch was a project that I started many years ago (in my colleague days) as a productive hobbie that would give me some speed at coding and a tool that I would use to make my work a little easier. With the time, I guess, it became something a little bigger than I expected. What is completely clear for me, It is not only my project anymore, but a community driven one.

My feeling is that the times when a "simple command line tool" are no longer enough have finally came. So, the question is; isn't it time to directly move on to a server and daemonized application and invest all the energies on that approach?

I'd like to know what users and contributors think about it.

Regards,
Mauro

**Comment by shelld3v** [comment]


**Reply by maurosoria** [reply]
I mean, instead of a CLI tool, a server app (REST) which you can interact with a client (could be a CLI or WebApp) where you can launch all the tasks (instead of a python process per scan, multiple scans simultaneously).

**Comment by valentijnscholten** [comment]
I think the `-l` option to provide a filename with urls to scan is nice. What a good addition would be if `dirsearch` could run multiple scans in parallel. Often you can only run a limited amount of threads per url, but running 10 scans for different urls in parallel would be no problem.
